src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -693,17 +695,20 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 
 	private async onTableClick(event: ITableMouseEvent) {
 		// account for not having the number column
-		let column = this.resultSet.columnInfo[event.cell.cell - 1];
+		const column = this.resultSet.columnInfo[event.cell.cell - 1];
 		// handle if a showplan link was clicked
-		if (column && (column.isXml || column.isJson)) {
+		if (column) {
 			const subset = await this.getRowData(event.cell.row, 1);
-			let value = subset[0][event.cell.cell - 1];
-			let content = value.displayValue;
-			const input = this.untitledEditorService.create({ mode: column.isXml ? 'xml' : 'json', initialValue: content });
-			await input.resolve();
-			await this.instantiationService.invokeFunction(formatDocumentWithSelectedProvider, input.textEditorModel, FormattingMode.Explicit, Progress.None, CancellationToken.None);
-			input.setDirty(false);
-			await this.editorService.openEditor(input);
+			const value = subset[0][event.cell.cell - 1];
+			const isJson = isJsonCell(value);
```

Comment:
Can you instead stick a property on the cell to tell us whether it's JSON or not? Avoid having to recheck the cell every time it's clicked.

Reply:
slickgrid is not caching the value associated with a cell, so we have to retrieve it anyway.